packages/loopring_v3/contracts/aux/liquidity-providers/BasicLiquidityProvider.sol

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../agents/FastWithdrawalAgent.sol";
+import "../../lib/Claimable.sol";
+import "../../lib/EIP712.sol";
+import "../../lib/ERC20.sol";
+import "../../lib/ERC20SafeTransfer.sol";
+import "../../lib/MathUint.sol";
+import "../../lib/ReentrancyGuard.sol";
+import "../../lib/SignatureUtil.sol";
+
+
+/// @title Basic contract storing funds for a liquidity provider.
+/// @author Brecht Devos - <brecht@loopring.org>
+contract BasicLiquidityProvider is ReentrancyGuard, Claimable
+{
+    using AddressUtil       for address;
+    using AddressUtil       for address payable;
+    using ERC20SafeTransfer for address;
+    using MathUint          for uint;
+    using SignatureUtil     for bytes32;
+
+    struct FastWithdrawalApproval
+    {
+        address exchange;
+        address from;
+        address to;
+        address token;
+        uint96  amount;
+        uint32  storageID;
+        uint64  validUntil; // most significant 32 bits as block height, least significant 32 bits as block time
+        bytes   signature;  // provider's signature
+    }
+
+    bytes32 constant public FASTWITHDRAWAL_APPROVAL_TYPEHASH = keccak256(
+        "FastWithdrawalApproval(address exchange,address from,address to,address token,uint96 amount,uint32 storageID,uint64 validUntil)"
+    );
+
+    bytes32 public DOMAIN_SEPARATOR;
+
+    FastWithdrawalAgent public agent;
+
+    constructor(FastWithdrawalAgent _agent)
+    {
+        agent = _agent;
+        DOMAIN_SEPARATOR = EIP712.hash(EIP712.Domain("BasicLiquidityProvider", "1.0", address(this)));
+    }
+
+    // User initiated fast withdrawal
+    function execute(FastWithdrawalApproval calldata approval)
```

Comment:
I suggest we change this to `function execute1(FastWithdrawalApproval[] calldata approvals)` and comment that *this function can be used by LPs to perform fast withdrawals which can be accelerated by users with a higher gas price using the same function*.

Also, change the other function to something like `execute2` so tools can be made to monitor pending fast withrawal transactions simply using the function selector.

Reply:
> this function can be used by LPs to perform fast withdrawals which can be accelerated by users with a higher gas price using the same function.

This function should only be called by the user, LPs shouldn't use this function because it will do an additional signature check that is unnecessary when the LP calls it himself.

Reply:
I think my goal is that users or any third-parties can accelerate any pending fast withdrawal transfers. If LPs use `function execute(FastWithdrawalAgent.Withdrawal[] memory withdrawals)`, user-initiated acceleration will be impossible.